src/main/java/com/mentoring/ecommerce/adapter/in/web/ProductController.java

```diff
@@ -0,0 +1,26 @@
+package com.mentoring.ecommerce.adapter.in.web;
+
+import com.mentoring.ecommerce.adapter.in.web.mapper.ProductMapper;
+import com.mentoring.ecommerce.adapter.in.web.request.ProductReq;
+import com.mentoring.ecommerce.application.port.in.SaveProductUseCase;
+import com.mentoring.ecommerce.domain.Product;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/products")
```

Comment:
@Pelicer, in the dummy project, the strategy used to define the path was to specify it in each of the endpoints instead of defining it just once as done here (line 16).

**Ex:** 

- Current project code: `@RequestMapping("/api/products")`
- Dummy project code: `@PostMapping(path = "/grocery")` | `@GetMapping(path = "/grocery")`

There was some specific reason for that?

Reply:
@Aristevao , that was due to the client specifying themselves how they wanted the URLs to look like. In our case, we had a default path, such as 'groceries', and we would, then, define in each endpoint the specific mapping for that function. Your method works fine as well.